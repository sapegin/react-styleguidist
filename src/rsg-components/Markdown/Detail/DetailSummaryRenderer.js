import React from 'react';
import PropTypes from 'prop-types';

import Styled from 'rsg-components/Styled';

const styles = ({ space, color, fontSize, fontFamily }) => ({
	summary: {
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		color: color.base,
		marginBottom: space[1],
	},
});

export function DetailSummaryRenderer({ classes, children }) {
	return <summary className={classes.summary}>{children}</summary>;
}
DetailSummaryRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
};

export default Styled(styles)(DetailSummaryRenderer);
