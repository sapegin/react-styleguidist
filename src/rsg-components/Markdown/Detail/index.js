export { default } from 'rsg-components/Markdown/Detail/DetailRenderer';
export { default as DetailSummary } from 'rsg-components/Markdown/Detail/DetailSummaryRenderer';
