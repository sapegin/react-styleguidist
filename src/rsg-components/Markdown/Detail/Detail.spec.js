import React from 'react';

import Detail from './index';

describe('Markdown Detail', () => {
	it('should render a Detail', () => {
		const actual = render(<Detail />);

		expect(actual).toMatchSnapshot();
	});
});
