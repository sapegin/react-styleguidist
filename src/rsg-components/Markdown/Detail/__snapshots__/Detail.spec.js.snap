// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Detail should render a Detail 1`] = `
<details
  class="rsg--details-0"
/>
`;
